Celeste.Mod.mm/Patches/GameLoader.cs

```diff
@@ -149,5 +151,37 @@ class patch_GameLoader : GameLoader {
                 return new OverworldLoader(startMode, snow);
             }
         }
+
+        /// <summary>
+        /// Initializes mod save data backups if required (if Everest was just updated past 2104).
+        /// </summary>
+        private static void checkModSaveDataBackups() {
+            Version previousVersion;
+            try {
+                previousVersion = new Version(CoreModule.Settings.CurrentVersion);
+            } catch {
+                // oops, version is null or can't be parsed for any reason.
+                previousVersion = new Version(0, 0, 0);
+            }
+
+            if (previousVersion < new Version(1, 2104, 0)) {
+                // user just upgraded: create mod save data backups.
+                // (this is very similar to OverworldLoader.CheckVariantsPostcardAtLaunch)
+                Logger.Log("core", $"User just upgraded from version {previousVersion}: creating mod save data backups.");
+
+                for (int i = 0; i < 3; i++) { // only the first 3 saves really matter.
+                    if (!UserIO.Exists(SaveData.GetFilename(i))) {
+                        continue;
+                    }
+                    SaveData saveData = UserIO.Load<SaveData>(SaveData.GetFilename(i), backup: false);
+                    saveData.AfterInitialize();
+                    if (saveData != null) {
+                        UserIO.Save<ModSaveData>(SaveData.GetFilename(saveData.FileSlot) + "-modsavedata", UserIO.Serialize(new ModSaveData(saveData as patch_SaveData)));
+                    }
```

Comment:
You're calling AfterInitialize on saveData even if it might be null. Also, what speaks against using BeforeSave?

Reply:
Calling BeforeSave() without actually saving felt weird 😕 I just tried calling it, but it crashed because orig_BeforeSave() tried accessing SaveData.Instance.